src/common.f90

```diff
@@ -34,7 +34,53 @@ module m_common
 
 contains
 
-  integer function get_rdr_from_dirs(dir_from, dir_to) result(rdr_dir)
+  pure subroutine get_dirs_from_rdr(dir_from, dir_to, rdr_dir)
+    integer, intent(out) :: dir_from, dir_to
+    integer, intent(in) :: rdr_dir
+    
+    select case (rdr_dir)
```

Comment:
What do you think about using `findloc` intrinsic in Fortran instead of the `select case` block? Haven't really experimented with this before but looks like we can use it to obtain `dir_from` and `dir_to` indices easily with a single query using the `rdr_map` array.

https://gcc.gnu.org/onlinedocs/gcc-12.3.0/gfortran/FINDLOC.html

Reply:
yeah, that's what I was wondering too. Performance wise I am not sure what's the best. The other option would be two lut for `from_dir` and `to_dir` of size 31 (with lots of zeros).

Reply:
You're right, it can be bad for performance, this one is called very often inside the reorder loop. Maybe its better to explore different options later when we're investigating the performance.

Reply:
actually, that call wasn't needed so I removed it. It should be better now performance wise, regardless of the implementation